sharding/nodesSetup_test.go

```diff
@@ -118,6 +119,89 @@ func TestNodesSetup_ProcessConfigNodesWithIncompleteDataShouldErr(t *testing.T)
 	require.Equal(t, ErrCouldNotParsePubKey, err)
 }
 
+func TestNodesSetup_ProcessConfigNodesShouldErrCouldNotParsePubKeyForString(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 2
+	ns := &NodesSetup{
+		addressPubkeyConverter: mock.NewPubkeyConverterMock(32),
+		validatorPubkeyConverter: &mock.PubkeyConverterMock{
+			DecodeCalled: func() ([]byte, error) {
+				return nil, ErrCouldNotParsePubKey
+			},
+		},
+	}
```

Comment:
we should create the component with the constructor with `NewNodesSetup` since we want to simulate the way we might use the component from outside

Reply:
applicable to all tests

Reply:
wouldn't this lead to increasing the initializations of components that are not necessary for these specific tests?

Reply:
yes, we should initializing components that are not necessary (so use just simple mocks/stubs if they are not used in the test scenario) but we should initialize the component with the constructor

Reply:
understood
solved it